src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -99,6 +101,17 @@ public class CheckerTest extends AbstractModuleTestSupport {
     @TempDir
     public File temporaryFolder;
 
+    private File createTempFile(String prefix) throws IOException {
+        return createTempFile(prefix, ".tmp");
+    }
+
+    private File createTempFile(String prefix, String suffix) throws IOException {
+        final String name = Optional.ofNullable(prefix).orElse("")
+                + UUID.randomUUID()
+                + Optional.ofNullable(suffix).orElse(".tmp");
```

Comment:
Please remove ofNullable , our temp file should always have extension, it is good for windows.

Reply:
Even Linux dont care much on extension, people like extensions